content/courses/native-onchain-development/program-security.md

```diff
@@ -260,40 +292,45 @@ stakers.
 
 #### Integer overflow and underflow
 
-Rust integers have fixed sizes. This means they can only support a specific
-range of numbers. An arithmetic operation that results in a higher or lower
-value than what is supported by the range will cause the resulting value to wrap
-around. For example, a `u8` only supports numbers 0-255, so the result of
-addition that would be 256 would actually be 0, 257 would be 1, etc.
+One of the common pitfalls when working with integers in Rust (and in Solana
+programs) is handling integer overflow and underflow. Rust integers have fixed
+sizes and can only hold values within a certain range. When a value exceeds that
+range, it wraps around, leading to unexpected results.
 
-This is always important to keep in mind, but especially so when dealing with
-any code that represents true value, such as depositing and withdrawing tokens.
+For example, with a `u8` (which holds values between 0 and 255), adding 1 to 255
+results in a value of 0 (overflow). To avoid this, you should use checked math
+functions like
+[`checked_add()`](https://doc.rust-lang.org/std/primitive.u8.html#method.checked_add)
+and
+[`checked_sub()`](https://doc.rust-lang.org/std/primitive.u8.html#method.checked_sub):
 
 To avoid integer overflow and underflow, either:
 
 1. Have logic in place that ensures overflow or underflow _cannot_ happen or
-2. Use checked math like `checked_add` instead of `+`
+2. Use checked math like `checked_add()` instead of `+`
+
    ```rust
    let first_int: u8 = 5;
    let second_int: u8 = 255;
-   let sum = first_int.checked_add(second_int);
+   let sum = first_int.checked_add(second_int)
+    .ok_or(ProgramError::ArithmeticOverflow)?;
    ```
 
 ## Lab
 
-Let’s practice together with the Movie Review program we've worked on in
-previous lessons. No worries if you’re just jumping into this lesson without
-having done the previous lesson - it should be possible to follow along either
-way.
-
-As a refresher, the Movie Review program lets users store movie reviews in PDA
-accounts. Last lesson, we finished implementing the basic functionality of
-adding a movie review. Now, we'll add some security checks to the functionality
-we've already created and add the ability to update a movie review in a secure
-manner.
+In this lab, we will build upon the Movie Review program that allows users to
```

Comment:
This was in the old lesson too, but Github markdown seems to break here - syntax highlighting starts including 'in' and 'self' etc as keywords in regular text. Maybe check the previous code block is terminated properly.

Reply:
I tried to find if there is anything missing or breaking in markdown but I am not able to find any issue and even quotes are fine. Let me know what can be done and this is not getting reflected in the rich diff view.